python/huggingfaceserver/health_check.py

```diff
@@ -13,41 +13,88 @@
 # limitations under the License.
 
 import argparse
+import sys
+import time
+import os
+
 import ray
 import requests
-import sys
 from kserve.logging import logger
 
 
-def initialize_ray_cluster():
-    if not ray.is_initialized():  # Check if Ray is already initialized
-        ray.init(address="auto")
-        return "Ray initialized"
-    else:
-        return "Ray already initialized"
+def initialize_ray_cluster(ray_address="auto"):
+    try:
+        if ray.is_initialized():  # Check if Ray is already initialized
+            return
+        else:
+            ray.init(address=ray_address)
+            return
+    except Exception as e:
+        logger.error(f"Failed to initialize Ray: {e}")
+        sys.exit(1)
+
+
+def show_ray_cluster_status(ray_address="auto"):
+    initialize_ray_cluster(ray_address)
+    try:
+        resources = ray.cluster_resources()
+        logger.info("Cluster resources:", resources)
+    except Exception as e:
+        logger.error(f"Error getting Ray nodes status: {e}")
 
 
-def verify_status(result):
-    if result == "Healthy":
+def verify_status(result, probe_type):
+    if result in ("Healthy", True):
+        logger.info(f"{probe_type} Probe: Healthy")
         sys.exit(0)
     else:
+        logger.error(f"{probe_type} Probe: Unhealthy")
         sys.exit(1)
 
 
 # Function for startup check using Ray API
-def check_startup():
-    try:
-        initialize_ray_cluster()
-        logger.info("Ray is accessible")
+def check_registered_node_and_runtime_health(
+    pipeline_parallel_size, health_check_url, ray_address="auto"
+):
+    initialize_ray_cluster(ray_address)
+    # Check if the registered nodes count matches PIPELINE_PARALLEL_SIZE
+    check_registered_nodes_status = check_registered_nodes(
+        pipeline_parallel_size, ray_address
+    )
+
+    # Check if server health return 200
+    check_runtime_health_status = check_runtime_health(health_check_url)
+    logger.debug(
+        f"check_registered_nodes_status: {check_registered_nodes_status},check_runtime_health_status: {check_runtime_health_status}"
+    )
+
+    if (
+        check_registered_nodes_status == "Healthy"
+        and check_runtime_health_status == "Healthy"
+    ):
         return "Healthy"
-    except Exception as e:
-        logger.error(f"Ray is NOT accessible: {e}")
+    else:
+        return "Unhealthy"
```

Comment:
it is better to return boolean instead unless there is a third state

Reply:
Right. I was also worried about this, but it seemed like the method name and return value didn't match, so I did it like this, but I'll think about it some more and change it later.